miri's broken on latest nightly

Not a test failure (soundness bug), just nightly feature breakage.

See: https://github.com/tosc-rs/mnemos/actions/runs/8618127330/job/23619965911#step:4:68

```
 error[E0635]: unknown feature `stdsimd`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/portable-atomic-1.4.2/src/lib.rs:327:13
    |
327 |     feature(stdsimd)
    |             ^^^^^^^

```

We should probably be running our Miri tests on the pinned nightly toolchain to avoid this kind of bullshit?

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -113,20 +113,20 @@ jobs:
   # (test-host forth3) - run miri tests
   test-host-miri:
     runs-on: ubuntu-latest
-    name: cargo miri test (host)
+    name: cargo miri test --package forth3 (host)
     needs: check
     # TODO(eliza): only run this if forth3 changed?
     # needs: changed_paths
     # if: needs.changed_paths.outputs.should_skip != 'true'
     steps:
-    - name: install rust toolchain and miri
+    - name: install Miri etc
       run: |
-        rustup toolchain install nightly --component miri
-        cargo +nightly miri setup
+        rustup component add miri
+        cargo miri setup
     - uses: actions/checkout@v3
     - name: cargo miri test (forth3)
       run: |
-        cargo +nightly miri test \
+        cargo miri test \
           --package forth3 \
           --all-features
 

